### DISCUSSION TITLE
Enabling Rich Text Editing in ASP.NET Core Textareas

### DISCUSSION THREAD

**Discussion by Xce15** [dotnet/aspnetcore#53397] [General] [answered]
Can someone help me in activating the Text editor in Textarea?
whenever I do it it fails the validating stage and never saves the actual inputs from the text area?

**Comment by Xyncgas** [comment]
try use jsinterop for getting the input
use div to implement rich text editing instead of textarea by setting the attribute contenteditable="true"
you can set the attribute to "plaintext-only" to drop rich text support and use it like a text area

it's fun implementing a rich text editor
good lucks, I see market top players are using virtualizations for example visual studio code so contenteditable is not enough

**Comment by raghadmta** [comment, marked as answer]
What tools are you using for the text editor?

check the initialization of the tool and whether it is linked to the entity or not.
also, check the values in your corresponding action by setting a breakpoint maybe it was required but got to be sent as an empty value due to an issue in the text editor.

there are two famous text editors people usually use: 
1- [TinyMCE](https://www.tiny.cloud/tinymce/) - multiple ways to install the best one yet to install it as a Nuget package.
2- [CKEditor](https://ckeditor.com/)  - most people use CDN to install it.
